state/controller.go

```diff
@@ -174,7 +173,7 @@ func (st *State) checkSpaceIsAvailableToAllControllers(spaceName string) error {
 		if err != nil {
 			return errors.Annotate(err, "cannot get machine")
 		}
-		if _, ok := network.SelectAddressesBySpaces(m.Addresses(), space.NetworkSpace()); !ok {
+		if _, ok := m.Addresses().InSpaces(space.NetworkSpace()); !ok {
```

Comment:
Again, method on the type provides better "materialisation of expectations" for the OO developer, over package-public methods.